lib/logflare/logs/otel_trace.ex

```diff
@@ -95,7 +96,20 @@ defmodule Logflare.Logs.OtelTrace do
 
   defp extract_key_value(%KeyValue{key: key, value: nil}), do: {key, nil}
 
+  defp extract_key_value(%KeyValue{
+         key: key,
+         value: %AnyValue{value: {:array_value, %ArrayValue{values: values}}}
+       }) do
+    {key, Enum.map(values, &extract_key_value/1)}
```

Comment:
The `Enum.map` here actually extracts just values. Perhaps it's worth separating a `extract_value` function, like this:

```elixir
defp extract_key_value(%KeyValue{key: key, value: value}) do
  {key, extract_value(value)}
end

defp extract_value(%AnyValue{value: {:array_value, %ArrayValue{values: values}}}) do
  Enum.map(values, &extract_value/1)
end

defp extract_value(%AnyValue{value: {_type, value}}), do: value
defp extract_value(nil), do: nil
defp extract_value(value), do: value
```

I added the last clause, because currently you have `defp extract_key_value(v), do: v`, though I am not sure if it's actually necessary, is it ever something other than `%AnyValue{}`?

Reply:
checked the otel spec and seems like AnyValue only gets used for arbitrary data values (this situation), but it is a bit vague if this is always used in the protobuf payload. 
I'll keep the last clause as fallback in case there is an unknown edge case, should be clearer if it is needed once we have better tooling around the protobuf struct generation for tests, in current state it is very much observational as opposed to using the otel sdk internals directly to generate the pb struct.